dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/http2/GenericHttp2ServerTransportListener.java

```diff
@@ -86,6 +86,7 @@ protected Executor initializeExecutor(Http2Header metadata) {
         return new SerializingExecutor(executorSupport.getExecutor(metadata));
     }
 
+    @Override
     protected void doOnMetadata(Http2Header metadata) {
         if (metadata.isEndStream()) {
             if (!HttpMethods.supportBody(metadata.method())) {
```

Comment:
**Security Issue**: Missing authorization checks in the doOnMetadata method which processes HTTP/2 headers. This could potentially allow unauthorized access to sensitive operations based on metadata content. <br> **Fix**: Implement authorization checks to verify that the user has the necessary permissions to perform operations based on the metadata content. <br> **Code Suggestion**: 
 ```
 +    if (!authorizeUser(metadata)) {
 +        throw new SecurityException("Unauthorized metadata access");
 +    }
 protected void doOnMetadata(Http2Header metadata) {
     if (metadata.isEndStream()) {
         if (!HttpMethods.supportBody(metadata.method())) {
 ```